README.md

```diff
@@ -80,6 +83,147 @@ async nuxtServerInit({ commit }, { $sentry }) {
 }
 ```
 
+## Lazy Loading (on the client)
+
+> :warning: Please be aware that lazy loading could prevent some errors from being reported
+
+Set `lazy: true` in your module options to load Sentry lazily on the client. This will prevent Sentry from being included in your main bundle **but could result in some errors not being reported**.
+
+You can also pass a lazy config object in your module options, see below.
+
+### Injected properties
+
+#### `$sentry` (mocked)
+- Type: `Object`
+
+Normally `$sentry` would always refer to the `@sentry/browser` api. But if we lazy load Sentry this api wont be available until Sentry has loaded. If you don't want to worry about whether Sentry is loaded or not, a mocked Sentry api is injected into the Nuxt.js context that will execute all Sentry api calls once Sentry is loaded
+
+See: `injectMock` and `mockApiMethods` options below
+
+#### `$sentryReady`
+- Type `Function`
+
+This method returns a Promise which will be resolved once Sentry has been loaded. You could use this instead of mocking `$sentry`.
+
+Example usage:
+```js
+this.$sentryReady().then((sentry) => sentry.captureMessage('Erreur!'))
+// or
+(await this.$sentryReady()).captureMessage('Erreur!')
+```
+
+#### `$sentryLoad`
+- Type: `Function`
+
+> Only injected when `injectLoadHook: true`.
+
+The callback you need to call to indicate that you are ready to load Sentry.
+
+Example usage:
+```js
+  // layouts/default.vue
+  ...
+  mounted() {
+    // This will only load sentry once an error was thrown
+    // To prevent a chicken & egg issue, make sure to also
+    // set injectMock: true if you use this so the error
+    // that triggered the load will also be captured
+    this.errorListener = () => {
+      this.$sentryLoad()
+      window.removeEventListener('error', errorListener)
+    }
+    window.addEventListener('error', errorListener)
+  },
+  destroyed() {
+    window.removeEventListener('error', this.errorListener)
+  }
+```
+
+### Lazy Options
```

Comment:
I dont mind, one of the main reasons why I kept this separate was because I didnt want to think about how to display second order options ;)

So, do you have a suggestion for that? Feel free to just commit directly into this pr otherwise

Reply:
I'll play with it and push a change if I can come up with a good solution.

Reply:
Addressed with 0f8b060edfa8a2071d11d8a24cb7db97e18d8cca